routes/users.js

```diff
@@ -34,26 +34,29 @@ router.route('/login')
   });
 
 router.route('/register')
-  // kept failing with using DEFAULT for id, until the fourth try.
   .post((req, res) => {
     knex.raw('SELECT * FROM users WHERE email = ?', [req.body.email])
-      .then((result) => {
-        if (result.rowCount > 0){
-          throw { message : 'User already exists'};
-        }
-      })
-    //   .then().knex.raw(`INSERT INTO users (id, email, password, created_at, updated_at)\
-    //     VALUES\
-    //     (DEFAULT, \'${req.body.email}\', \'${req.body.password}\', DEFAULT, DEFAULT)`)
-    //   })
-    
-    // .then((result) => {
-    //   res.send(result);
-    // })
-    // .catch((err) => {
-    //   res.send(err);
-    // })
+    .then((result) => {
+      if (result.rowCount > 0){
+        res.status(404).json({ message : 'User already exists'});
+      } else {
+        return knex.raw(
+          `INSERT INTO users (id, email, password, created_at, updated_at)\
+          VALUES\
+          (DEFAULT, \'${req.body.email}\', \'${req.body.password}\', DEFAULT, DEFAULT)`);
+      }
+    })
+    .then ((result) => {
+      res.send(result);
```

Comment:
This should be moved into the then above and into the else conditional to prevent res.send from being called twice.